contrib/export/v034/gov/types.go

```diff
@@ -0,0 +1,126 @@
+package gov
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	"time"
+)
+
+type GenesisState struct {
+	StartingProposalID uint64                `json:"starting_proposal_id"`
+	Deposits           []DepositWithMetadata `json:"deposits"`
+	Votes              []VoteWithMetadata    `json:"votes"`
+	Proposals          []Proposal            `json:"proposals"`
+	DepositParams      DepositParams         `json:"deposit_params"`
+	VotingParams       VotingParams          `json:"voting_params"`
+	TallyParams        TallyParams           `json:"tally_params"`
+}
+
+type DepositWithMetadata struct {
+	ProposalID uint64  `json:"proposal_id"`
+	Deposit    Deposit `json:"deposit"`
+}
+
+type VoteWithMetadata struct {
+	ProposalID uint64 `json:"proposal_id"`
+	Vote       Vote   `json:"vote"`
+}
+
+type Proposal struct {
+	Content `json:"content"` // Proposal content interface
+
+	ProposalID       uint64             `json:"proposal_id"`        //  ID of the proposal
+	Status           gov.ProposalStatus `json:"proposal_status"`    // Status of the Proposal {Pending, Active, Passed, Rejected}
+	FinalTallyResult TallyResult        `json:"final_tally_result"` // Result of Tallys
+
+	SubmitTime     time.Time `json:"submit_time"`      // Time of the block where TxGovSubmitProposal was included
+	DepositEndTime time.Time `json:"deposit_end_time"` // Time that the Proposal would expire if deposit amount isn't met
+	TotalDeposit   sdk.Coins `json:"total_deposit"`    // Current deposit on this proposal. Initial value is set at InitialDeposit
+
+	VotingStartTime time.Time `json:"voting_start_time"` // Time of the block where MinDeposit was reached. -1 if MinDeposit is not reached
+	VotingEndTime   time.Time `json:"voting_end_time"`   // Time that the VotingPeriod for this proposal will end and votes will be tallied
+}
+
+type Proposals []Proposal
+
+type Deposit struct {
+	ProposalID uint64         `json:"proposal_id"` //  proposalID of the proposal
+	Depositor  sdk.AccAddress `json:"depositor"`   //  Address of the depositor
+	Amount     sdk.Coins      `json:"amount"`      //  Deposit amount
+}
+
+type Deposits []Deposit
+
+type Vote struct {
+	ProposalID uint64         `json:"proposal_id"` //  proposalID of the proposal
+	Voter      sdk.AccAddress `json:"voter"`       //  address of the voter
+	Option     gov.VoteOption `json:"option"`      //  option from OptionSet chosen by the voter
```

Comment:
I tried to use the copied type but I keep getting errors, do I have to register more types on the codec?

Reply:
Possibly? Can you please post/share the exact error you get? It's probably because the type(s) need to be registered.

Reply:
Sure
```
panic: json: cannot unmarshal string into Go value of type gov.ProposalStatus

goroutine 1 [running]:
github.com/tendermint/go-amino.(*Codec).MustUnmarshalJSON(...)
        /Users/sabau/.go/pkg/mod/github.com/tendermint/go-amino@v0.15.0/amino.go:445
github.com/cosmos/cosmos-sdk/contrib/export/v036.Migrate(0xc00015c630, 0xc00018cbd0, 0x7ffeefbff5b9)
        /Users/sabau/.go/src/github.com/cosmos/cosmos-sdk/contrib/export/v036/migrate.go:80 +0x1126
main.main()
        /Users/sabau/.go/src/github.com/cosmos/cosmos-sdk/contrib/export/main.go:66 +0x2aa
exit status 2

```

Reply:
identical if I remove it from gov.VoteOption

Reply:
Fixed.